### PR TITLE
Sync typeshed

### PR DESCRIPTION
**Pull request by JukkaL**
Source commit:
https://github.com/python/typeshed/commit/2445eddb4b67fdaa58ec7c2113ff1542021a6206

### CODE REVIEW

**Comment by sobolevn** [comment]
After this is merged, we can rerun https://github.com/python/mypy/pull/11585 to make sure it is not failing anymore.
This PR should fix `ClassVar` problem we had: https://github.com/python/typeshed/pull/6348

**Comment by JukkaL** [comment]
I checked some of the new messages, and the new errors look legitimate -- mostly about `__file__` possibly being `None`.

**Comment by hauntsaninja** [comment]
Legitimate, sure, but I checked and the `__file__` warnings are in practice false positives: https://github.com/python/typeshed/pull/6186#issuecomment-949937412

**Comment by JukkaL** [comment]
In some cases where optional types have caused false positives, we've used types like `str | Any`. I don't have a strong opinion on this. In any case any change should be made to typeshed first.

**Comment by intgr** [comment]
This PR seems to cause regression #11646.

File: mypy/typeshed/stdlib/typing.pyi [hunk 1]
```diff
@@ -435,29 +431,29 @@ class Mapping(Collection[_KT], Generic[_KT, _VT_co]):
...
     # see discussion in https://github.com/python/typing/pull/273.
     @abstractmethod
-    def __getitem__(self, k: _KT) -> _VT_co: ...
+    def __getitem__(self, __k: _KT) -> _VT_co: ...
     # Mixin methods
     @overload
     def get(self, key: _KT) -> _VT_co | None: ...
     @overload
-    def get(self, key: _KT, default: _VT_co | _T) -> _VT_co | _T: ...
+    def get(self, __key: _KT, __default: _VT_co | _T) -> _VT_co | _T: ...
```
Comment (intgr) [inline mypy/typeshed/stdlib/typing.pyi:439]: OK I see, this is the cause of #11646.

The `default=` argument might not apply to every `Mapping.get`, but it does work with `dict`. So maybe there should be special handling for `dict.get`?

Comment (intgr) [reply mypy/typeshed/stdlib/typing.pyi:439]: Doh. My bad, it `dict.get()` does not support keyword arguments, the original code was instead using `QueryDict` from Django.

So I suppose this should be solved in django-stubs instead.

**Pull request merged**